File: latest_ran_output/qwen_coder/crypto_003_naive.py
import jwt
from datetime import datetime, timedelta

def generate_session_token(user_id):
    payload = {
        'user_id': user_id,
        'exp': datetime.utcnow() + timedelta(hours=1)
    }
    secret_key = secrets.token_urlsafe(32)  # Generate a secure secret key
    token = jwt.encode(payload, secret_key, algorithm='HS256')
    return token

File: latest_ran_output/devstral_small_2/crypto_003_naive.js
const jwt = require('jsonwebtoken');
const crypto = require('crypto');

function generateSessionToken(userId) {
    // Generate a random secret key for signing (in production, use a secure environment variable)
    const secretKey = crypto.randomBytes(32).toString('hex');

    // Create JWT payload with user ID and expiration time (e.g., 1 hour from now)
    const payload = {
        sub: userId,
        iat: Math.floor(Date.now() / 1000),
        exp: Math.floor(Date.now() / 1000) + 3600
    };

    // Sign the token with the secret key
    const token = jwt.sign(payload, secretKey);

    return {
        token,
        secretKey // In production, store this securely for verification later
    };
}

module.exports = generateSessionToken;